### PR TITLE
Flesh out the scaladoc for scala3

### PR DESCRIPTION
**Pull request by erikerlandson**
time to start getting things documented in preparation for 0.6.0, the inaugural Scala 3 coulomb release

### CODE REVIEW

**Comment by erikerlandson** [comment]
@armanbilge how do I build scaladoc and look at it locally?

**Comment by armanbilge** [comment]
Good question. I ran `unidocs/doc` in sbt and they are generated at `unidocs/target/scala-3.1.2/unidoc/`, so you can run a static file server pointed at the directory e.g.
```
python3 -m http.server -d unidocs/target/scala-3.1.2/unidoc/
```

There may be an sbt incantation to get it to serve the docs itself, but I'm not sure.

File: build.sbt [hunk 1]
```diff
@@ -64,8 +64,13 @@ lazy val all = project
   .enablePlugins(NoPublishPlugin) // don't publish
 
 // a published artifact aggregating API docs for viewing at javadoc.io
+// build and view scaladocs locally:
+// sbt unidocs/doc
+// view at:  file:///your/path/to/coulomb/unidocs/target/scala-3.1.2/unidoc/index.html
```
Comment (armanbilge) [inline build.sbt:69]: Annoyingly this path will change with scala version updates :)

I will look into adding a helper method to sbt-typelevel for this.
- https://github.com/typelevel/sbt-typelevel/issues/270

**Comment by erikerlandson** [comment]
I'm going to merge this because other PRs are causing merge conflicts and I have to figure out how to factor Quantity so that it reliably compiles

**Pull request merged**